### PR TITLE
Release 0.2.1 (official release)

### LINKED ISSUE #88: Release 0.2.0
# Description
Do a bug bush to make sure all features are working properly. Specifiacially we need to test without `kedro-viz` installation in environment (or have a incompatible version in environment, i.e. pydantic 1.0).

0.2.0
Release after next kedro viz (1st week of September, as rely on both the main npm/python `kedro-viz`)

# Checklist
- [ ] #117
- [ ] #115
- [ ] #114
- [ ] Update Telemetry HEAP ID
- [ ] #91 (Pin 10.0.0) Update kedro-viz version after latest release
- [ ] #93 Docs
- [x] https://github.com/kedro-org/kedro-viz/issues/1968
- [x] https://github.com/kedro-org/vscode-kedro/issues/72
- [x] #68
- [x] https://github.com/kedro-org/vscode-kedro/issues/62
- [x] https://github.com/kedro-org/vscode-kedro/pull/65
- [x] https://github.com/kedro-org/kedro-viz/issues/2022
- [x] #56 
- [x] https://github.com/kedro-org/kedro-viz/issues/2024
- [x] https://github.com/kedro-org/vscode-kedro/pull/70
- [x] https://github.com/kedro-org/kedro-viz/pull/2049
- [x] https://github.com/kedro-org/kedro-viz/pull/2060
- [x] https://github.com/kedro-org/vscode-kedro/pull/92
- [x] https://github.com/kedro-org/vscode-kedro/pull/104
- [x] https://github.com/kedro-org/vscode-kedro/pull/99
- [x] #71

### PR DESCRIPTION
**Pull request by noklam**
# How to install
fix #88 

you can install pre-release version of extension from marketplace directly:
- Click the arrow next to the `Install` button
image

# How to test
- Make sure you open VSCode at the root of a Kedro project
- Make sure VSCode is using the correct virtual environment, i.e. the environment that you can do `kedro run`
- Bonus: Start a fresh new environment, kedro project to test because we need to make sure dependencies is not a problem. We added a new flow in case user are missing certain dependencies.
- Test Kedro: Run Viz
- playaround with namespace, non spaceflights project structure.

- #71

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # Change Log
 
-
 # 0.2.0
+## Major changes and new features
+- Integrated Kedro-Viz flowchart into the extension with a new command `Kedro: Run Viz`.
```
Comment (jitu5) [inline CHANGELOG.md:5]:
```suggestion
- Integrated Kedro-Viz flowchart into the extension with a new command `Kedro: Run Kedro Viz`.
```

**Comment by noklam** [comment]
> @noklam Will it be useful to track who is installing and canceling "Kedro-Viz" dependency in telemetry ?

We cannot track who is installing since that's personal information. I think it's more important to know if people are cancelling because of bug or it doesn't work, the usage of the installation is not too important since we are not going to remove it simply because people are not clicking on it.

**Comment by noklam** [comment]
One thing that I want to improve though, is a little bit more help or at least guide user to report the issue when the server didn't start properly. I think that's the most frustrating part for now, since it doesn't do anything and user have little information about what's going on.

Based on some previous chat a few users have installed it but it didn't worked for their project so they didn't use the extension. Sometimes it's simply because they are not opening the workspace or missing dependencies to start the extension. (This is separated and we don't need to block this for the release, just something to consider in the future).

**Comment by jitu5** [comment]
> One thing that I want to improve though, is a little bit more help or at least guide user to report the issue when the server didn't start properly. I think that's the most frustrating part for now, since it doesn't do anything and user have little information about what's going on.
> 
> Based on some previous chat a few users have installed it but it didn't worked for their project so they didn't use the extension. Sometimes it's simply because they are not opening the workspace or missing dependencies to start the extension. (This is separated and we don't need to block this for the release, just something to consider in the future).

On similar line we also need to fix those [panic errors](https://github.com/kedro-org/vscode-kedro/issues/77) when not on a Kedro project.

**Comment by noklam** [comment]
Good point, I want to spare some time after this feature release on more bug fix, UX improvement and start adding some tests.

#58 
image

**Comment by astrojuanlu** [comment]
Right after opening my IDE I found this error:

```
2024-09-20 17:50:45.543 [info] Traceback (most recent call last):
  File "/Users/juan_cano/.vscode/extensions/kedro.kedro-0.2.0/bundled/tool/lsp_server.py", line 75, in <module>

2024-09-20 17:50:45.543 [info]     import yaml
ModuleNotFoundError: No module named 'yaml'
```

https://github.com/kedro-org/vscode-kedro/blob/8ed1f3054632407405d2d7bb2b9635811e9c3d60/bundled/tool/lsp_server.py#L75

> Make sure VSCode is using the correct virtual environment, i.e. the environment that you can do kedro run

It was indeed solved by setting the correct Python interpreter.

> Test Kedro: Run Viz

I get the "missing dependency" prompt, click "Install", I get an info message saying "Kedro-Viz and its Dependencies installed successfully!", and then nothing happens ❓ 

I run Run Viz again and I get the same prompt.

Looking at the output, I get "No module named pip", which makes a ton of sense because my environment doesn't have pip installed (`uv venv` doesn't install pip by default).

I eventually solved it by doing `uv add kedro-viz` myself.

I now see a black screen though, and nothing meaningful in the output.

image

**Comment by jitu5** [comment]
> Right after opening my IDE I found this error:
> 
> ```
> 2024-09-20 17:50:45.543 [info] Traceback (most recent call last):
>   File "/Users/juan_cano/.vscode/extensions/kedro.kedro-0.2.0/bundled/tool/lsp_server.py", line 75, in <module>
> 
> 2024-09-20 17:50:45.543 [info]     import yaml
> ModuleNotFoundError: No module named 'yaml'
> ```
> 
> https://github.com/kedro-org/vscode-kedro/blob/8ed1f3054632407405d2d7bb2b9635811e9c3d60/bundled/tool/lsp_server.py#L75
> 
> > Make sure VSCode is using the correct virtual environment, i.e. the environment that you can do kedro run
> 
> It was indeed solved by setting the correct Python interpreter.
> 
> > Test Kedro: Run Viz
> 
> I get the "missing dependency" prompt, click "Install", I get an info message saying "Kedro-Viz and its Dependencies installed successfully!", and then nothing happens ❓
> 
> I run Run Viz again and I get the same prompt.
> 
> Looking at the output, I get "No module named pip", which makes a ton of sense because my environment doesn't have pip installed (`uv venv` doesn't install pip by default).
> 
> I eventually solved it by doing `uv add kedro-viz` myself.
> 
> I now see a black screen though, and nothing meaningful in the output.
> 
> image

@astrojuanlu About the blank screen, LSP server error log coming from Kedro-Viz was missing and just print statement was added, we will fix this before final release. In one of my Kedro project I also got blank screen, on debug I found "Class 'pandas.CSVDataset' not found" error. Which kedro project are you using ?

**Comment by astrojuanlu** [comment]
I was using a WIP version of https://github.com/astrojuanlu/kedro-pypi-monitor, not sure if you'll be able to reproduce. Let me know when I can try a version with logs.

**Comment by jitu5** [comment]
> I was using a WIP version of https://github.com/astrojuanlu/kedro-pypi-monitor, not sure if you'll be able to reproduce. Let me know when I can try a version with logs.

@astrojuanlu I tried and able to reproduce same blank screen and in log I got below error. I have a extension with log, you have to install it.

```
Globals key 'start_date' not found and no default value provided.
    full_key: pypi_kedro_raw.sql
    object_type=dict
```

**Comment by astrojuanlu** [comment]
Thanks, looks like problem was between chair and keyboard then!

Any chance to show an error message instead of a blank screen? Or at least an error notification like "couldn't display Kedro Viz, check logs"

**Comment by noklam** [comment]
> Looking at the output, I get "No module named pip", which makes a ton of sense because my environment doesn't have pip installed (uv venv doesn't install pip by default).

That was indeed the assumptions with pip. [c189778](https://github.com/kedro-org/vscode-kedro/pull/121/commits/c18977834ff1b22aa0cf725db28afbe0afaf15dd) Update the message slightly.

For the blank screen, any idea @jitu5?

**Review by rashidakanchwala — COMMENTED** [review]
I am still reviewing some other things. 
But I also got a blank Kedro-viz screen initially, and Nok showed me how I need to go to `Outputs` on VSCode, select `Kedro` to understand what error was being thrown. Not sure, if others will also face the same issue and know where to find the error logs.

**Comment by jitu5** [comment]
> couldn't display Kedro Viz, check logs

@noklam @astrojuanlu 
In case of blank screen or server error while getting json data by Kedro-Viz, I can show error message like this,
Screenshot 2024-09-23 at 10 43 59 a m

**Comment by jitu5** [comment]
> looks good for me, I notice this seems to be a common thing to do among extensions, maybe it's even better make this one step instead of Output -> Kedro? We will need to update the README.md to reflect the troubleshoot section too. image
> 
> They all implement a simple command that directly go to the extension outputs, for example: https://github.com/microsoft/vscode/blob/464f63c2f7ec85dfa7c1946ba62994072945cef2/extensions/tunnel-forwarding/src/extension.ts#L99-L106

@noklam If we can directly open Kedro output check with command, So I make a link, by clicking on that user will directly jump to Kedro output logs

error-log

**Comment by noklam** [comment]
That's look smooooth

**Review by astrojuanlu — APPROVED** [review]
Ship ittttttt

**Review by rashidakanchwala — APPROVED** [review]
This is suppper exciting!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+- Modify the extension to search pipelines from all <package_name>/pipelines folder.
```
[the remark this answers is not in the document]
Comment (noklam) [reply CHANGELOG.md]: done

**Comment by noklam** [comment]
Updated release title to be 0.2.1 since the VSCode extension treat pre-release as an official release and not allow the same version name.

**Comment by noklam** [comment]
Updated the contribution point so this is now a public facing command `Kedro: Show logs` to simplify the troubleshooting steps.

**Pull request merged**